Canvas/canvas_scraper.py

```
@@ -0,0 +1,98 @@
+        path = os.path.join(output_dir, filename)
+        with open(path, 'wb') as f:
+            for chunk in resp.iter_content(chunk_size=8192):
+                if chunk:
+                    f.write(chunk)
+        return path
+
+
+def _demo() -> None:
+    base = "https://learning.unisg.ch"
+    scraper = CanvasScraper(base)
+    module_html = open(os.path.join("Canvas", "Canvas_Corporate_Finance_Module_html_F12_export.html"), encoding="utf-8").read()
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)